In [1]:
import warnings
warnings.filterwarnings('ignore')

Firstly, we download necessary libraries, upload scraped data and then display the top rows of all tables right away. To remark, we possess six datasets each representing either and the certain job posting site's information in the USA

In [2]:
import pandas as pd
import numpy as np

In [3]:
USA_glassdoor = pd.read_excel(r"C:\Users\mmatr\OneDrive\Рабочий стол\JobsProject\JobsData\USA\GlassDoor\Glassdoor_Data_analyst_Jobs_USA.xlsx")
USA_indeed = pd.read_excel(r"C:\Users\mmatr\OneDrive\Рабочий стол\JobsProject\JobsData\USA\Indeed\Indeed_USA_Job_listing.xlsx")
USA_linkedin = pd.read_excel(r"C:\Users\mmatr\OneDrive\Рабочий стол\JobsProject\JobsData\USA\LinkedIn\LinkedIn_Data Analyst_jobs_United_States_main.xlsx")

In [4]:
display(USA_glassdoor.head(2))
display(USA_indeed.head(2))
display(USA_linkedin.head(2))

job_title      company_name  \
0  Data Entry Analyst - Remote /Office  Angelo Elite Inc   
1                     Jr. Data Analyst        Net2Aspire   

                                         joblink_URL        location  \
0  https://www.glassdoor.ca/partner/jobListing.ht...  Washington, DC   
1  https://www.glassdoor.ca/partner/jobListing.ht...          Remote   

                       salary_range  rating  emp_type  job_level  \
0  US$23.00 Per Hour(Employer Est.)     NaN       NaN        NaN   
1   US$65K - US$80K (Employer Est.)     NaN       NaN        NaN   

   job_responsib  industry  
0            NaN       NaN  
1            NaN       NaN

job_title        company_name  \
0           Quality Analyst              TikTok   
1  Jr Data Analyst - Remote  UnitedHealth Group   

                                         joblink_URL  \
0  https://www.indeed.com/rc/clk?jk=b7c710aba8c1a...   
1  https://www.indeed.com/rc/clk?jk=bf2ff874cd368...   

                           location  salary_range  rating  emp_type  \
0        Nashville, TN\n+1 location           NaN     NaN       NaN   
1  Remote in Eden Prairie, MN 55346           NaN     NaN       NaN   

   job_level  job_responsib  industry  
0        NaN            NaN       NaN  
1        NaN            NaN       NaN

job_title   company_name  \
0  Data Analyst  Alan Gray LLC   
1  Data Analyst    RennerBrown   

                                         joblink_URL                 location  \
0  https://www.linkedin.com/jobs/view/data-analys...               Boston, MA   
1  https://www.linkedin.com/jobs/view/data-analys...  New York, United States   

          salary_range  rating         job_level   emp_type  \
0                  NaN     NaN       Entry level  Full-time   
1  $115,000 - $130,000     NaN  Mid-Senior level  Full-time   

            job_responsib                       industry  
0  Information Technology                      Insurance  
1  Information Technology  IT Services and IT Consulting

We ensured that dataframes' columns match, we can concatenate those pieces into one single table

In [5]:
usa_jobs = pd.concat([USA_glassdoor, USA_indeed, USA_linkedin])

Straightaway we secure our data entries from being mistakenly repeated and for both tables we apply method drop_duplicates by subset of job title, company's name and location attributes in order to preserve the fairness and quality of future analysis

In [6]:
display('Before: '+str(usa_jobs.shape))
usa_jobs = usa_jobs.drop_duplicates(subset = ['job_title', 'company_name', 'location'])
display('After: '+str(usa_jobs.shape))

'Before: (3152, 10)'

'After: (1757, 10)'

First of all, we will take a look at data type of each column, verify its correctness and examine the overall description:

In [7]:
usa_jobs.dtypes

job_title         object
company_name      object
joblink_URL       object
location          object
salary_range      object
rating           float64
emp_type          object
job_level         object
job_responsib     object
industry          object
dtype: object

In [8]:
usa_jobs.describe(include='all')

job_title   company_name  \
count           1757           1754   
unique           992           1256   
top     Data Analyst  SynergisticIT   
freq             417             60   
mean             NaN            NaN   
std              NaN            NaN   
min              NaN            NaN   
25%              NaN            NaN   
50%              NaN            NaN   
75%              NaN            NaN   
max              NaN            NaN   

                                              joblink_URL location  \
count                                                1754     1757   
unique                                               1754      903   
top     https://www.glassdoor.ca/partner/jobListing.ht...   Remote   
freq                                                    1      160   
mean                                                  NaN      NaN   
std                                                   NaN      NaN   
min                                                   NaN      NaN   
25%                                                   NaN      NaN   
50%                                                   NaN      NaN   
75%                                                   NaN      NaN   
max                                                   NaN      NaN   

                           salary_range      rating   emp_type    job_level  \
count                               223  127.000000        682          684   
unique                              200         NaN          5            6   
top     US$65K - US$80K (Employer Est.)         NaN  Full-time  Entry level   
freq                                  5         NaN        402          445   
mean                                NaN    3.922047        NaN          NaN   
std                                 NaN    0.565252        NaN          NaN   
min                                 NaN    2.000000        NaN          NaN   
25%                                 NaN    3.600000        NaN          NaN   
50%                                 NaN    3.900000        NaN          NaN   
75%                                 NaN    4.300000        NaN          NaN   
max                                 NaN    5.000000        NaN          NaN   

                 job_responsib                       industry  
count                      682                            682  
unique                      42                             89  
top     Information Technology  IT Services and IT Consulting  
freq                       536                            276  
mean                       NaN                            NaN  
std                        NaN                            NaN  
min                        NaN                            NaN  
25%                        NaN                            NaN  
50%                        NaN                            NaN  
75%                        NaN                            NaN  
max                        NaN                            NaN

In [9]:
usa_jobs[usa_jobs['company_name'] == 'SynergisticIT']

job_title   company_name  \
171                                 Junior Data Analyst  SynergisticIT   
268                                 Junior Data Analyst  SynergisticIT   
270                                 Junior Data Analyst  SynergisticIT   
271                          Junior Data Analyst-Remote  SynergisticIT   
273                                 Junior Data Analyst  SynergisticIT   
274                                 Junior Data Analyst  SynergisticIT   
276                                 Junior Data Analyst  SynergisticIT   
279                                 Junior Data Analyst  SynergisticIT   
280                                 Junior Data Analyst  SynergisticIT   
287                          Junior Data Analyst-Remote  SynergisticIT   
331                                 Junior Data Analyst  SynergisticIT   
336                                 Junior Data Analyst  SynergisticIT   
345                                 Junior Data Analyst  SynergisticIT   
347                            Entry level data analyst  SynergisticIT   
348                            Entry Level Data Analyst  SynergisticIT   
350                            Entry Level Data Analyst  SynergisticIT   
395                          Junior Data Analyst-Remote  SynergisticIT   
398                                 Junior Data Analyst  SynergisticIT   
406                                 Data Analyst-Remote  SynergisticIT   
413                            Entry Level Data Analyst  SynergisticIT   
416                            Entry Level Data Analyst  SynergisticIT   
460                            Entry Level Data Analyst  SynergisticIT   
468                                 Junior Data Analyst  SynergisticIT   
470                                 Junior Data Analyst  SynergisticIT   
471                                 Junior Data Analyst  SynergisticIT   
473                            Entry Level data Analyst  SynergisticIT   
474                               Data Analyst (Remote)  SynergisticIT   
477                                 Junior Data Analyst  SynergisticIT   
483                            Entry Level Data Analyst  SynergisticIT   
532                                 Junior Data Analyst  SynergisticIT   
535                                 Junior Data Analyst  SynergisticIT   
539                            Entry Level Data Analyst  SynergisticIT   
543                            Entry Level Data Analyst  SynergisticIT   
544                                 Data Analyst-Remote  SynergisticIT   
596                            Entry Level Data Analyst  SynergisticIT   
600                                        Data Analyst  SynergisticIT   
622                            Entry level Data Analyst  SynergisticIT   
632                                        Data Analyst  SynergisticIT   
686                            ENTRY LEVEL DATA ANALYST  SynergisticIT   
687                                        Data Analyst  SynergisticIT   
691                                        Data Analyst  SynergisticIT   
692                                        Data Analyst  SynergisticIT   
710                                        Data Analyst  SynergisticIT   
712                                        Data Analyst  SynergisticIT   
721                                        Data Analyst  SynergisticIT   
729                                        Data Analyst  SynergisticIT   
732                                        Data Analyst  SynergisticIT   
746                                        Data Analyst  SynergisticIT   
747   Entry Level Data Analyst/ Python programmer po...  SynergisticIT   
752                             Entry Level Data Anlyst  SynergisticIT   
757                                        Data Analyst  SynergisticIT   
803                                        Data Analyst  SynergisticIT   
804                                        Data Analyst  SynergisticIT   
859                                        Data An

Splendid, types of all columns matches the needs of analysis they will be used for. The exception is 'salary_range' field, but the whole section will be devoted for its cleaning and normalizing.

The 'describe' method also did not discover any issues. The amount of unique titles and company names differ from their total amount, because there are cases when one company offers different jobs. For example SynergisticIT, the company appearing the most frequently, is looking for staff to fill 19 distinct positions in different locations all over the US.

Let's begin defining the site the job offer is posted on, relying on the fields we have in access

In [10]:
usa_jobs['site'] = usa_jobs['joblink_URL'].apply(lambda x: 'linkedin' if 'linkedin' in str(x).lower() else ('indeed' if 'indeed' in str(x).lower() else 'glassdoor')) 

I suggest to proceed with working out the 'location' field:

In [11]:
#complete 'remote' feature engineering
usa_jobs['remote_locat'] = usa_jobs['location'].apply(lambda x: 'yes' if 'remote' in x.lower() else 'no')
usa_jobs['remote_job_tit'] = usa_jobs['job_title'].apply(lambda x: 'yes' if 'remote' in x.lower() else 'no')
usa_jobs['remote'] = usa_jobs.apply(lambda x: 'yes' if x['remote_job_tit'] == 'yes' else ('yes' if x['remote_locat'] == 'yes' else 'no'), axis=1)
usa_jobs.drop(columns = ['remote_locat', 'remote_job_tit'], axis=1,inplace=True)


In [12]:
#pull out the city and state
usa_jobs['city'] = usa_jobs['location'].apply(lambda x:x.split('\n')[0].split(',')[0])
usa_jobs['state'] = usa_jobs['location'].apply(lambda x: x.split('\n')[0].split(',')[1][:3] if ',' in x else 'not indicated')

Still there are some misperfections in the newly derived fields, such as 'remote' and 'United States' in 'city' column, however, these values are quite informative and can be handled easily during the exploratory analysis and visualization phases. Therefore we are done with these attributes

Next stage is to get familiar with salary_range values:

In [13]:
usa_jobs['salary_range'].unique()

array(['US$23.00 Per Hour(Employer Est.)',
       'US$65K - US$80K (Employer Est.)',
       'US$11.64 - US$31.10 Per Hour(Employer Est.)', nan,
       'US$47.00 - US$55.00 Per Hour(Employer Est.)',
       'US$25.00 - US$30.00 Per Hour(Employer Est.)',
       'US$40K - US$60K (Employer Est.)',
       'US$55.00 - US$70.00 Per Hour(Employer Est.)',
       'US$41K - US$57K (Glassdoor Est.)',
       'US$28K - US$39K (Glassdoor Est.)',
       'US$35K - US$40K (Employer Est.)',
       'US$20.00 - US$24.00 Per Hour(Employer Est.)',
       'US$33K - US$43K (Glassdoor Est.)',
       'US$85K - US$95K (Employer Est.)',
       'US$50K - US$60K (Employer Est.)',
       'US$60K - US$70K (Employer Est.)',
       'US$30.00 - US$45.00 Per Hour(Employer Est.)',
       'US$32.00 - US$42.00 Per Hour(Employer Est.)',
       'US$55.00 - US$58.00 Per Hour(Employer Est.)',
       'US$80.00 - US$90.00 Per Hour(Employer Est.)',
       'US$68K - US$107K (Glassdoor Est.)',
       'US$43K - US$61K (Glassdoor Est.)'

As we can see there two noticeable patterns of values of the column

It is necessary to mark certain rows with no salary provided, so we are able to slice the data, based on this predicate

In [14]:
usa_jobs['salary_range'] = usa_jobs['salary_range'].fillna('-1')

At first, we filter out the rows, where salary is not provided, then replace all capital letters "K" after the number with three zeroes to indicate thousands, remove dollar sign before actual numbers and truncate the text part at all

In [15]:
clean_salary_usa = usa_jobs[usa_jobs['salary_range'] != '-1']

In [16]:
#Ensure there is no missing values
display('Number of missing values: ' + str(clean_salary_usa['salary_range'].isna().sum()))

'Number of missing values: 0'

In [17]:
clean_salary_usa['salary_range'] = clean_salary_usa['salary_range'].apply(lambda x:x.replace('K', '000').replace('US$', '').replace('$', '').replace(',','').replace('£','').split('(')[0].split(' P')[0])

In [18]:
clean_salary_usa['salary_range'].unique()

array(['23.00', '65000 - 80000 ', '11.64 - 31.10', '47.00 - 55.00',
       '25.00 - 30.00', '40000 - 60000 ', '55.00 - 70.00',
       '41000 - 57000 ', '28000 - 39000 ', '35000 - 40000 ',
       '20.00 - 24.00', '33000 - 43000 ', '85000 - 95000 ',
       '50000 - 60000 ', '60000 - 70000 ', '30.00 - 45.00',
       '32.00 - 42.00', '55.00 - 58.00', '80.00 - 90.00',
       '68000 - 107000 ', '43000 - 61000 ', '55.00 - 60.00',
       '54000 - 83000 ', '19.00 - 26.00', '56000 - 79000 ',
       '58000 - 86000 ', '70000 - 109000 ', '63000 - 98000 ',
       '40.00 - 50.00', '48000 - 70000 ', '60000 - 94000 ', '70.00',
       '63000 - 93000 ', '20.00 - 25.00', '130000 - 140000 ',
       '43000 - 79000 ', '48000 - 52000 ', '48000 - 110000 ', '20.00',
       '55000 - 93000 ', '65.00 - 70.00', '17.75 - 19.25',
       '55000 - 79000 ', '12.00 - 14.00', '50000 - 70000 ',
       '58000 - 84000 ', '28.00 - 35.00', '90000 - 127000 ',
       '20.00 - 22.00', '39000 - 55000 ', '55.00 - 65.00', '28.00',
 

In [19]:
clean_salary_usa['min_salary'] = clean_salary_usa['salary_range'].apply(lambda x:x.split('-')[0].strip())
clean_salary_usa['min_salary'] = clean_salary_usa['min_salary'].astype('float')

In [20]:
clean_salary_usa['max_salary_test'] = clean_salary_usa['salary_range'].apply(lambda x:x[-7:])
clean_salary_usa['max_salary_test2'] = clean_salary_usa['max_salary_test'].apply(lambda x:x.replace('-','').strip())
clean_salary_usa['max_salary'] = clean_salary_usa['max_salary_test2'].apply(lambda x: x[-3:].strip() if ' ' in x else x.strip())
clean_salary_usa['max_salary'] = clean_salary_usa['max_salary'].astype('float')
clean_salary_usa.drop(columns = ['max_salary_test', 'max_salary_test2'], axis = 1, inplace=True)

In [21]:
#converting from hour wage to year salary, taking 2080 as an amount of working hours in 2023
clean_salary_usa['max_salary'] = clean_salary_usa['max_salary'].apply(lambda x: x*2080 if x<150 else x)
clean_salary_usa['min_salary'] = clean_salary_usa['min_salary'].apply(lambda x: x*2080 if x<150 else x)

Just in case, we drop duplicates once more

In [22]:
display('Shape before dropping duplicates once more: '+str(clean_salary_usa.shape))
clean_salary_usa = clean_salary_usa.drop_duplicates(subset = ['job_title', 'company_name'])
display('Shape after dropping duplicates the second time: '+str(clean_salary_usa.shape))

'Shape before dropping duplicates once more: (223, 16)'

'Shape after dropping duplicates the second time: (215, 16)'

In [23]:
clean_salary_usa.describe()

rating     min_salary     max_salary
count  104.000000     215.000000     215.000000
mean     3.950962   75608.204651   96091.559070
std      0.573186   36724.129280   40592.001213
min      2.000000       0.000000       0.000000
25%      3.600000   50000.000000   67820.000000
50%      4.000000   65000.000000   90000.000000
75%      4.300000  100000.000000  120000.000000
max      5.000000  208000.000000  260000.000000

In general, all fields contain adequate values. The only suspicious part is minimal value of max and min salary. Let's take a closer look at it:

In [24]:
clean_salary_usa.sort_values('max_salary').head()

job_title  \
1188            Sector Data Analyst: Energy/Industrials   
1045                                       Data Analyst   
1179                           RESEARCH DATA ANALYST II   
53    Entry Level Data Analyst - Automotive Data Ana...   
10                 Criminal Data Entry Analyst (REMOTE)   

                            company_name  \
1188      Balyasny Asset Management L.P.   
1045                         TechDigital   
1179  California Department of Education   
53                     Geneva Management   
10                        InfoMart, Inc.   

                                            joblink_URL        location  \
1188  https://www.linkedin.com/jobs/view/sector-data...    New York, NY   
1045  https://www.linkedin.com/jobs/view/data-analys...     Detroit, MI   
1179  https://www.linkedin.com/jobs/view/research-da...  Sacramento, CA   
53    https://www.glassdoor.ca/partner/jobListing.ht...      Reston, VA   
10    https://www.glassdoor.ca/partner/jobListing.ht...    Marietta, GA   

        salary_range  rating   emp_type         job_level  \
1188           0 - 0     NaN  Full-time  Mid-Senior level   
1045           0 - 0     NaN   Contract       Entry level   
1179     5793 - 7256     NaN  Full-time    Not Applicable   
53     12.00 - 14.00     3.0        NaN               NaN   
10    28000 - 39000      3.3        NaN               NaN   

                        job_responsib                       industry  \
1188  Other, Engineering, and Finance          Investment Management   
1045           Information Technology  IT Services and IT Consulting   
1179           Information Technology                      Insurance   
53                                NaN                            NaN   
10                                NaN                            NaN   

           site remote        city state  min_salary  max_salary  
1188   linkedin     no    New York    NY         0.0         0.0  
1045   linkedin     no     Detroit    MI         0.0         0.0  
1179   linkedin     no  Sacramento    CA      5793.0      7256.0  
53    glassdoor    yes      Reston    VA     24960.0     29120.0  
10    glassdoor    yes    Marietta    GA     28000.0     39000.0

There are three rows, missing the meaningful value for salary column. In order to fix this we are going to replace it with NaN values, so there will be no misinterpretation during EDA and visualization stages

In [25]:
clean_salary_usa['min_salary'] = clean_salary_usa['min_salary'].apply(lambda x: x if x != 0 else np.nan)
clean_salary_usa['max_salary'] = clean_salary_usa['max_salary'].apply(lambda x: x if x != 0 else np.nan)

In [26]:
clean_salary_usa.describe()

rating     min_salary     max_salary
count  104.000000     212.000000     213.000000
mean     3.950962   76678.132075   96993.827230
std      0.573186   35852.588527   39690.398785
min      2.000000    5793.000000    7256.000000
25%      3.600000   50000.000000   70000.000000
50%      4.000000   65000.000000   90000.000000
75%      4.300000  100000.000000  120000.000000
max      5.000000  208000.000000  260000.000000

In [27]:
clean_salary_usa.sort_values('min_salary').head()

job_title  \
1179                           RESEARCH DATA ANALYST II   
2                              Entry Level Data Analyst   
53    Entry Level Data Analyst - Automotive Data Ana...   
10                 Criminal Data Entry Analyst (REMOTE)   
14                                   Data Entry Analyst   

                            company_name  \
1179  California Department of Education   
2                         Talentheed Inc   
53                     Geneva Management   
10                        InfoMart, Inc.   
14                    OLM Food Solutions   

                                            joblink_URL        location  \
1179  https://www.linkedin.com/jobs/view/research-da...  Sacramento, CA   
2     https://www.glassdoor.ca/partner/jobListing.ht...          Remote   
53    https://www.glassdoor.ca/partner/jobListing.ht...      Reston, VA   
10    https://www.glassdoor.ca/partner/jobListing.ht...    Marietta, GA   
14    https://www.glassdoor.ca/partner/jobListing.ht...     Milford, IA   

        salary_range  rating   emp_type       job_level  \
1179     5793 - 7256     NaN  Full-time  Not Applicable   
2      11.64 - 31.10     NaN        NaN             NaN   
53     12.00 - 14.00     3.0        NaN             NaN   
10    28000 - 39000      3.3        NaN             NaN   
14    33000 - 43000      2.7        NaN             NaN   

               job_responsib   industry       site remote        city  \
1179  Information Technology  Insurance   linkedin     no  Sacramento   
2                        NaN        NaN  glassdoor    yes      Remote   
53                       NaN        NaN  glassdoor    yes      Reston   
10                       NaN        NaN  glassdoor    yes    Marietta   
14                       NaN        NaN  glassdoor     no     Milford   

              state  min_salary  max_salary  
1179             CA      5793.0      7256.0  
2     not indicated     24211.2     64688.0  
53               VA     24960.0     29120.0  
10               GA     28000.0     39000.0  
14               IA     33000.0     43000.0

As it turns out there is one posting, where the salary is per month. Let's change it:

In [28]:
clean_salary_usa['min_salary'] = clean_salary_usa['min_salary'].apply(lambda x: x*12 if (x > 5000 and x < 10000) else x)
clean_salary_usa['max_salary'] = clean_salary_usa['max_salary'].apply(lambda x: x*12 if (x > 5000 and x < 10000) else x)

In [29]:
clean_salary_usa.describe()

rating     min_salary     max_salary
count  104.000000     212.000000     213.000000
mean     3.950962   76978.712264   97368.550235
std      0.573186   35521.073799   39213.081444
min      2.000000   24211.200000   29120.000000
25%      3.600000   50000.000000   70000.000000
50%      4.000000   65000.000000   90000.000000
75%      4.300000  100000.000000  120000.000000
max      5.000000  208000.000000  260000.000000

We join the cleaned salary data with the rest of records

In [30]:
cleaned_usa_df = usa_jobs.merge(clean_salary_usa, on=['job_title','company_name'], how='outer', suffixes=['', '_clean'])

In [31]:
columns_to_drop = ['joblink_URL_clean', 'location_clean',
       'salary_range_clean', 'rating_clean', 'emp_type_clean',
       'job_level_clean', 'job_responsib_clean', 'industry_clean',
       'site_clean', 'remote_clean', 'city_clean', 'state_clean']
cleaned_usa_df = cleaned_usa_df.drop(columns_to_drop, axis=1)

In [32]:
cleaned_usa_df.columns

Index(['job_title', 'company_name', 'joblink_URL', 'location', 'salary_range',
       'rating', 'emp_type', 'job_level', 'job_responsib', 'industry', 'site',
       'remote', 'city', 'state', 'min_salary', 'max_salary'],
      dtype='object')

In [33]:
#I added column which indicates what country is the job from for the convenience's sake
#during future union with Canada's jobs dataset
cleaned_usa_df['Country'] = 'USA'

In [34]:
cleaned_usa_df.head()

job_title  \
0              Data Entry Analyst - Remote /Office   
1                                 Jr. Data Analyst   
2                         Entry Level Data Analyst   
3  Data Entry Analyst - HSCSN - Remote Opportunity   
4                                     Data Analyst   

                 company_name  \
0            Angelo Elite Inc   
1                  Net2Aspire   
2              Talentheed Inc   
3  The HSC Health Care System   
4            Bainbridge, Inc.   

                                         joblink_URL        location  \
0  https://www.glassdoor.ca/partner/jobListing.ht...  Washington, DC   
1  https://www.glassdoor.ca/partner/jobListing.ht...          Remote   
2  https://www.glassdoor.ca/partner/jobListing.ht...          Remote   
3  https://www.glassdoor.ca/partner/jobListing.ht...   United States   
4  https://www.glassdoor.ca/partner/jobListing.ht...          Remote   

                                  salary_range  rating emp_type job_level  \
0             US$23.00 Per Hour(Employer Est.)     NaN      NaN       NaN   
1              US$65K - US$80K (Employer Est.)     NaN      NaN       NaN   
2  US$11.64 - US$31.10 Per Hour(Employer Est.)     NaN      NaN       NaN   
3                                           -1     3.5      NaN       NaN   
4                                           -1     3.6      NaN       NaN   

  job_responsib industry       site remote           city          state  \
0           NaN      NaN  glassdoor    yes     Washington             DC   
1           NaN      NaN  glassdoor    yes         Remote  not indicated   
2           NaN      NaN  glassdoor    yes         Remote  not indicated   
3           NaN      NaN  glassdoor    yes  United States  not indicated   
4           NaN      NaN  glassdoor    yes         Remote  not indicated   

   min_salary  max_salary Country  
0     47840.0     47840.0     USA  
1     65000.0     80000.0     USA  
2     24211.2     64688.0     USA  
3         NaN         NaN     USA  
4         NaN         NaN     USA

With intent to clean the data from copies across the postings from different websites, we drop duplicates by 'job_title', 'company_name', 'city' columns.

In [35]:
display('Before: '+str(cleaned_usa_df.shape))
cleaned_usa_df = cleaned_usa_df.drop_duplicates(subset= ['job_title', 'company_name', 'city'])
display('After: '+str(cleaned_usa_df.shape))

'Before: (1757, 17)'

'After: (1742, 17)'

We have a dataset consisted of 1742 cleaned and verified records about jobs from all over the USA ready to be exported.

In [36]:
#cleaned_usa_df.to_csv(r"C:/Users/mmatr/OneDrive/Рабочий стол/JobsProject/Cleaned_DFs/ready_USAJobs_df_excel")